src/com/person/Person.java

```diff
@@ -0,0 +1,75 @@
+package com.person;
+
+import com.enumex.EGender;
+import com.enumex.EStatus;
+
+import java.util.Date;
+import java.util.UUID;
+
+public class Person {
+
+    public static final String TO_STRING = "%s,%s,%s,%s,%s,%s";
+
+    private UUID uuid;
+    private String name;
+    private Date dateOfBirth;
+    private String interests;
+    private EGender gender;
+    private EStatus status;
+
+    public Person(String name, EGender gender, Date dateOfBirth, String interests, EStatus status) {
+        this.uuid = UUID.randomUUID();
+        this.name = name;
+        this.gender = gender;
+        this.dateOfBirth = dateOfBirth;
+        this.interests = interests;
+        this.status = status;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Date getDateOfBirth() {
+        return dateOfBirth;
+    }
+
+    public String getInterests() {
+        return interests;
+    }
+
+    public EGender getGender() {
+        return gender;
+    }
+
+    public EStatus getStatus() {
+        return status;
+    }
+
+    public UUID getUuid() {
+        return uuid;
+    }
+
+    @Override
+    public String toString() {
+        return String.format(TO_STRING, uuid, name, gender, dateOfBirth, interests, status);
+    }
+
+    @Override
+    public int hashCode() {
+        return this.uuid.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if(this == obj) {
+            return true;
+        }
+
+        if(obj instanceof Person) {
+            Person objectPerson = (Person) obj;
+            return this.uuid == objectPerson.getUuid();
```

Comment:
What if the person has everything the same but the uuid different? Does that mean that it is different person?

Reply:
I believe yes, If two persons share everything but uuid. There are not equals

Reply:
No! Think about it like it is a real object and a real person. A real person uuid is his ID number for example, but that does  not mean that he cant have same names, date of birth, interests,gender etc.. The equals method should check everything EXCEPT the uuid here! Same goes for hashcode